File: caesar_cipher.py
ALPHABET = ' ABCDEFGHIJKLMNOPQRSTUVWXYZ'

def encrypt(free_text, key):
    cipher_text = ''
    for c in free_text.upper():
        index = ALPHABET.find(c);
        index = (index + key) % len(ALPHABET)
        cipher_text = cipher_text + ALPHABET[index]
    return cipher_text

def decrypt(cipher_text, key):
    free_text = ''
    for c in cipher_text:
        index = ALPHABET.find(c)
        index = (index - key) % len(ALPHABET)
        free_text = free_text + ALPHABET[index]
    return free_text

if __name__ == '__main__':
    cipher_text = encrypt('I am COOL', 3)
    print(decrypt(cipher_text, 3))
    

File: Course_material/Java/CaesarCipher/src/com/globalsoftwaresupport/CaesarCipher.java
package com.globalsoftwaresupport;

public class CaesarCipher {
	
	// there are 26+1 characters
	private String ALPHABET = " ABCDEFGHIJKLMNOPQRSTUVWXYZ";
	
	public String encrypt(String plainText, int key) {
		
		String cipherText = "";
		// we make the algorithm case insensitive
		plainText = plainText.toUpperCase();
		
		// consider all the letters in the plain_text
		for(int i=0;i<plainText.length();i++) {
			char character = plainText.charAt(i);
			// find the numerical representation (index) associated with 
			// that character in the alphabet
			int charIndex = ALPHABET.indexOf(character); 
			// Caesar encryption is just a simple shift of characters according to the key
			// use modular arithmetic to transform the values within the range 
			// [0,num_of_letters_in_alphabet]
			int encryptedIndex = Math.floorMod(charIndex+key, ALPHABET.length());
			// keep appending the encrypted character to the cipher_text
			cipherText += ALPHABET.charAt(encryptedIndex);
		}
		
		return cipherText;	
	}
	
	public String decrypt(String cipherText, int key) {
		
		//the input cipherText is already upper case
		String plainText = "";
		
		for(int i=0;i<cipherText.length();++i) {
			char character = cipherText.charAt(i);
			int charIndex = ALPHABET.indexOf(character);
			int decryptedIndex = Math.floorMod(charIndex-key, ALPHABET.length());
			plainText += ALPHABET.charAt(decryptedIndex);
		}
		
		return plainText;
	}
}







